internal/backend/runtime/omni/controllers/omni/machine_set_node.go

```diff
@@ -184,12 +219,14 @@ func (ctrl *MachineSetNodeController) reconcileMachineSet(
 	machineStatusMap map[resource.ID]*omni.MachineStatus,
 	visited map[resource.ID]struct{},
 	logger *zap.Logger,
-) error {
-	var err error
-
+) (requiredAdditionalMachines int, err error) {
 	spec := machineSet.TypedSpec()
 	if spec.Value.MachineClass == nil || machineSet.Metadata().Phase() == resource.PhaseTearingDown {
-		return nil
+		if err = r.Destroy(ctx, omni.NewMachineSetRequiredMachines(resources.DefaultNamespace, machineSet.Metadata().ID()).Metadata()); err != nil && !state.IsNotFoundError(err) {
```

Comment:
Artem knows better, idk if we need some extra finalizers to handle this?

Reply:
This controller is of the old type, so I guess we don't need finalizers here.

But it may be necessary to handle `Teardown` + `Destroy` flow if `MachineSetRequiredMachines` resource gets some finalizers.

Reply:
Removed `Destroy` from this controller as Artem's suggestion, so no more applicable.